public/index.php

```diff
@@ -43,6 +43,11 @@
 require_once SYSTEMPATH . 'Config/DotEnv.php';
 (new CodeIgniter\Config\DotEnv(ROOTPATH))->load();
 
+// Define ENVIRONMENT
+if (! defined('ENVIRONMENT')) {
+    define('ENVIRONMENT', env('CI_ENVIRONMENT', 'production'));
+}
+
```

Comment:
Duplicate code. I would like to suggest the following changes for index.php and spark.
1. Move the php version check to the top of the bootstrap.php file.
2. Move the code below to the bottom of the bootstrap.php file.
```diff
-- require_once SYSTEMPATH . 'Config/DotEnv.php';
(new CodeIgniter\Config\DotEnv(ROOTPATH))->load();

// Define ENVIRONMENT
if (! defined('ENVIRONMENT')) {
    define('ENVIRONMENT', env('CI_ENVIRONMENT', 'production'));
}

// Grab our CodeIgniter
$app = Config\Services::codeigniter();
$app->initialize();
```
`require_once SYSTEMPATH . 'Config/DotEnv.php';` This line doesn't make sense, since in bootstrap.php we've already initiated the autoloader.

Reply:
The php version check should be in the top of index.php.
Because future PHP syntax change will cause errors  before the check.

Reply:
I would like to hear another person's opinion on moving the DotEnv process to the bootstrap.

Reply:
We've discussed the PHP check before and it does need to stay. 

It seems to make sense that DotEnv would be handled in bootstrap - I can't think of what the down side is to that?

Reply:
Should DotEnv be handling the environment setting explicitly? I've been frustrated in the past that it isn't available until after `CodeIgniter` is instantiated. Here is my workaround in my container:
```php
    private static function detectEnvironment(): string
    {
        if (defined('ENVIRONMENT')) {
            return ENVIRONMENT;
        }
        if (isset($_ENV['CI_ENVIRONMENT'])) {
            return $_ENV['CI_ENVIRONMENT'];
        }
        if (isset($_SERVER['CI_ENVIRONMENT'])) {
            return $_SERVER['CI_ENVIRONMENT'];
        }
        if ($env = getenv($_SERVER['CI_ENVIRONMENT'])) {
            return $env;
        }

        return 'production';
    }
```
I think if we ensure DotEnv handles the environment and `ENVIRONMENT` is defined in bootstrap it would always be available in our system classes.

Reply:
> It seems to make sense that DotEnv would be handled in bootstrap - I can't think of what the down side is to that?

If I don't use `DotEnv` at all, now I can get rid of it from `index.php`.
But if we move it to `bootstrap.php`, `DotEnv` will be always instantiated even if I don't use.

Reply:
I would like to exclude from this PR the question "Do we move the code from `index.php` to `bootstrap.php`".

Reply:
Agreed. Please create an issue to track this conversation.

Reply:
Done. #7824